test/frontend-integration-test/helloworld.spec.js

```diff
@@ -182,6 +184,90 @@ describe('deploy helloworld sample run', () => {
       return logs.indexOf(outputParameterValue + ' from node: ') > -1;
     }, waitTimeout);
   });
+
+  it('navigates back to the experiment list', () => {
+    $('button=Experiments').click();
+    browser.waitUntil(() => {
+      return new URL(browser.getUrl()).hash.startsWith('#/experiments');
+    }, waitTimeout);
+  });
+
+  it('creates a new run without selecting an experiment', () => {
+    $('#createNewRunBtn').waitForVisible();
+    $('#createNewRunBtn').click();
+
+    $('#choosePipelineBtn').waitForVisible();
+    $('#choosePipelineBtn').click();
+
+    $('.tableRow').waitForVisible();
+    $('.tableRow').click();
+
+    $('#usePipelineBtn').click();
+
+    $('#pipelineSelectorDialog').waitForVisible(waitTimeout, true);
+
+    browser.keys('Tab');
+    browser.keys(runWithoutExperimentName);
+
+    browser.keys('Tab');
+    browser.keys(runWithoutExperimentDescription);
+
+    // Skip over "choose experiment" button
+    browser.keys('Tab');
+
+    browser.keys('Tab');
+    browser.keys(outputParameterValue);
+
+    // Deploy
+    $('#startNewRunBtn').click();
+  });
+
+  it('redirects back to all runs page', () => {
+    browser.waitUntil(() => {
+      return (new URL(browser.getUrl())).hash === '#/runs';
+    }, waitTimeout, `URL was: ${new URL(browser.getUrl())}`);
+  });
+
+  it('displays both runs in all runs page', () => {
+    let attempts = 30;
+
+    // Wait for a reasonable amount of time until the run starts
+    while (attempts && $$('.tableRow').length !== 2) {
+      browser.pause(1000);
+      if (browser.isVisible('#refreshBtn')) {
+        $('#refreshBtn').click();
+      }
+      --attempts;
+    }
+
+    assert(attempts, `waited for 30 seconds but table had ${$$('.tableRow').length} entries. URL was: ${browser.getUrl()}`);
+  });
+
+  it('navigates back to the experiment list', () => {
+    $('button=Experiments').click();
+    browser.waitUntil(() => {
+      return new URL(browser.getUrl()).hash.startsWith('#/experiments');
+    }, waitTimeout);
+  });
+
+  it('displays both custom and default experiment experiment list page', () => {
+    $('.tableRow').waitForVisible();
+    const rows = $$('.tableRow').length;
+    assert(rows === 2, 'there should now be two experiments in the table, instead there are: ' + rows);
```

Comment:
I'm curious, does the default experiment disappear if this experiment-less run goes away? What if it's archived for example?

Reply:
Nope. Experiments can exist without runs